### PR TITLE
Create Rule “use-one-version-manager-supporting-multiple-softwares/rule"

### PR DESCRIPTION
**Pull request by johnxu77**

### CODE REVIEW

File: rules/use-one-version-manager-supporting-multiple-softwares/rule.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+Use one version manager supporting multiple softwares.
+
+<!--endintro-->
+
+![](ring.png)
+
+Every programming language or framework build upon it has its own version managers in community, such as NVM for Node.js, RVM for Ruby, and Pyenv for Python.
+
+However, it is not a good idea to use different version managers for different softwares. It is better to use one version manager supporting multiple softwares.
```
Comment (tiagov8) [inline rules/use-one-version-manager-supporting-multiple-softwares/rule.md:25]: 1. Please fix

Screenshot 2023-09-04 at 1 23 19 PM

File: rules/use-one-version-manager-supporting-multiple-softwares/rule.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+authors:
+- title: John Xu
+  url: https://ssw.com.au/people/john-xu
+related: []
+redirects:
+- do-you-know-to-use-one-version-manager-supporting-multiple-softwares
+
+---
+
+Use one version manager supporting multiple softwares.
```
Comment (tiagov8) [inline rules/use-one-version-manager-supporting-multiple-softwares/rule.md:17]: 1. I would have a better intro

See what I got from ChatGPT:

> write an intro for 
> 
> Use one version manager supporting multiple softwares.
> 
> ChatGPT
> "Managing software versions can be a complex and time-consuming task, especially when you're dealing with multiple applications across various platforms. In today's fast-paced digital landscape, having a unified solution that supports multiple software packages is invaluable. Imagine a single version manager that streamlines the process, ensuring that all your software remains up to date effortlessly. In this article, we explore the advantages and possibilities of using a version manager capable of handling multiple software applications, simplifying your workflow and enhancing your software management experience."

File: rules/use-one-version-manager-supporting-multiple-softwares/rule.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+✅ Easy to use  
+✅ Lightweight & Fast  
+✅ Non-intrusive  
+✅ [~700 registered plugins covering most use cases](https://github.com/asdf-vm/asdf-plugins) including *node, dotnet, dotnet-core*  
+
+❌ Linux/Mac/WSL only
+
+`youtube: https://www.youtube.com/embed/RTaqWRj-6Lg`  
+**Figure: Use asdf to manage versions of Python, NodeJS, GoLang and more!**
```
Comment (tiagov8) [inline rules/use-one-version-manager-supporting-multiple-softwares/rule.md:42]: 1. Change from "Figure:" to "Video:"
2. Add the time length at the end (https://www.ssw.com.au/rules/right-format-to-write-videos-time-length/)

File: rules/use-one-version-manager-supporting-multiple-softwares/rule.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+Every programming language or framework build upon it has its own version managers in community, such as NVM for Node.js, RVM for Ruby, and Pyenv for Python.
+
+However, it is not a good idea to use different version managers for different softwares. It is better to use one version manager supporting multiple softwares.
+
+
+### Option A: asdf (recommended)
+
+![](asdf.png)
+
+[asdf - manage all your runtime versions with one tool](https://asdf-vm.com/#/)
```
Comment (tiagov8) [inline rules/use-one-version-manager-supporting-multiple-softwares/rule.md:32]: 1. Add link only on the software name "asdf"
2. Make the 1st letter Uppercase after the dash

File: rules/use-one-version-manager-supporting-multiple-softwares/rule.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+❌ Linux/Mac/WSL only
+
+`youtube: https://www.youtube.com/embed/RTaqWRj-6Lg`  
+**Figure: Use asdf to manage versions of Python, NodeJS, GoLang and more!**
+
+### Option B: nix
+
+![](nix.png)
+
+[nix-shell - a tool that allows you to create and manage isolated, reproducible development environments for software projects, using the Nix package manager and a declarative configuration file](https://nixos.org/#asciinema-demo-cover)
```
Comment (tiagov8) [inline rules/use-one-version-manager-supporting-multiple-softwares/rule.md:48]: 1. Add link only on the software name "nix-shell"
2. Make the 1st letter Uppercase after the dash

File: rules/use-one-version-manager-supporting-multiple-softwares/rule.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+`youtube: https://www.youtube.com/embed/5Dd7rQPNDT8`  
+**Figure: How to use Nix on Ubuntu or any Linux Distro**
```
Comment (tiagov8) [inline rules/use-one-version-manager-supporting-multiple-softwares/rule.md:51]: 1. Change from "Figure:" to "Video:"
2. Add the time length at the end (https://www.ssw.com.au/rules/right-format-to-write-videos-time-length/)

Comment (johnxu77) [reply rules/use-one-version-manager-supporting-multiple-softwares/rule.md]: Done

File: rules/use-one-version-manager-supporting-multiple-softwares/rule.md (hunk 1 above)
Comment (johnxu77) [reply rules/use-one-version-manager-supporting-multiple-softwares/rule.md]: Done - fix these references

File: rules/use-one-version-manager-supporting-multiple-softwares/rule.md (hunk 2 above)
Comment (johnxu77) [reply rules/use-one-version-manager-supporting-multiple-softwares/rule.md]: Wow, this intro is really great

File: rules/use-one-version-manager-supporting-multiple-softwares/rule.md (hunk 4 above)
Comment (johnxu77) [reply rules/use-one-version-manager-supporting-multiple-softwares/rule.md]: Done

File: rules/use-one-version-manager-supporting-multiple-softwares/rule.md (hunk 5 above)
Comment (johnxu77) [reply rules/use-one-version-manager-supporting-multiple-softwares/rule.md]: Done

**Review by tiagov8 — APPROVED** [review]

**Review by tiagov8 — COMMENTED** [review]
LGTM

**Pull request merged**